pkg/server/endpoints/node/handler.go

```diff
@@ -201,13 +201,21 @@ func (h *Handler) Attest(stream node.Node_AttestServer) (err error) {
 		return status.Error(codes.Internal, "failed to compose response")
 	}
 
-	isAttested, err := h.isAttested(ctx, agentID)
+	isAttested, node, err := h.isAttested(ctx, agentID)
 	switch {
 	case err != nil:
 		log.WithError(err).Error("Failed to determine if agent has already attested")
 		return status.Error(codes.Internal, "failed to determine if agent has already attested")
 	case isAttested:
-		if err := h.updateAttestationEntry(ctx, svid[0]); err != nil {
+		req := &datastore.UpdateAttestedNodeRequest{
```

Comment:
Hmm, I wonder if Attest behavior should remain unchanged since the reason an agent is calling attest is because it has either no SVID, or the SVID is otherwise invalid. If the agent fails to persist the creds returned by attest, it would be re-attesting anyway.

Thoughts, @evan2645?

Reply:
I don't see any behavioral changes in this bit - I think this part of the diff is just adapting the call to update the node entry since its function signature has changed

Reply:
The behavioral change is that the SVID resulting from attestation is put in the "new" slot, allowing the old SVID to still be used until the new SVID is used to authenticate. I would have assumed that the new SVID would be set as the current SVID and the "new" fields zeroed...

Reply:
Ooh yes, I missed this... good catch @azdagron 

IMO attestation should stomp on any pre-existing stuff. I _think_ that means as you suggested, that the current svid field is set to the newly minted cert and the new svid field is zeroed

Reply:
Thanks for catching that! I'll fix it!